resource_retriever/src/retriever.cpp

```diff
@@ -111,6 +111,25 @@ size_t curlWriteFunc(void * buffer, size_t size, size_t nmemb, void * userp)
   return size * nmemb;
 }
 
+std::string Retriever::escape_spaces(const std::string & url)
```

Comment:
I don't think we need to make this part of the class; it never accesses anything in it, so it can just be a free function:

```suggestion
static std::string escape_spaces(const std::string & url)
```